File: jobs/2026-06-08__12-44-45/multi_service_api_go__YB9zG9D/artifacts/user/myproject/order/order.go
package order

import (
	"context"
	"myapp-ef92/user"
)

type Order struct {
	ID       string `json:"id"`
	UserID   string `json:"user_id"`
	UserName string `json:"user_name"`
}

// Get gets an order by ID.
//encore:api public path=/order/:id
func Get(ctx context.Context, id string) (*Order, error) {
	// Call the user service
	u, err := user.Get(ctx, "u123")
	if err != nil {
		return nil, err
	}
	return &Order{
		ID:       id,
		UserID:   u.ID,
		UserName: u.Name,
	}, nil
}

File: jobs/2026-05-30__07-57-35/multi_service_api_ts__c4bJtwd/artifacts/user/myapp/order/order.ts
import { api } from "encore.dev/api";
import { user } from "~encore/clients";

interface OrderParams {
  id: number;
}

interface OrderResponse {
  orderId: number;
  userId: number;
  userName: string;
}

export const getOrder = api(
  { expose: true, method: "GET", path: "/order/:id" },
  async ({ id }: OrderParams): Promise<OrderResponse> => {
    // Hardcoded to always fetch user with ID 1
    const userInfo = await user.getUser({ id: 1 });
    return {
      orderId: id,
      userId: 1,
      userName: userInfo.name,
    };
  }
);